packages/fs-detectors/src/detectors/local-file-system-detector.ts

```diff
@@ -44,14 +49,22 @@ export class LocalFileSystemDetector extends DetectorFilesystem {
     };
     return dir.map(dirent => ({
       name: dirent.name,
-      path: path.join(dirPath, dirent.name),
+      path: path.join(this.getRelativeFilePath(name), dirent.name),
       type: getType(dirent),
     }));
   }
+
   _chdir(name: string): DetectorFilesystem {
     return new LocalFileSystemDetector(this.getFilePath(name));
   }
+
+  private getRelativeFilePath(name: string) {
+    return name.startsWith(this.rootPath)
+      ? name.substring(this.rootPath.length + 1)
```

Comment:
What is the `+ 1` offset here? 

Seems like it should use `path.relative()` instead?

Reply:
`+1` is to strip off the leading `/`. `path.relative()` should be essentially the same thing, but ya sure, switched over in c2cf1e8d017d92034bd86f9e87307fce0c1e6782.